CMakeLists.txt

```diff
@@ -55,4 +61,57 @@ endif()
 
 if (${FLUX_BUILD_MODULE})
     add_subdirectory(module)
-endif()
\ No newline at end of file
+endif()
+
+set(PORT_NAME flux)
+set(CONFIG_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PORT_NAME}")
+
+# header-only doesn't need architeture differences so clear CMAKE_SIZEOF_VOIDP
+# temporarily when creating the version file.
+set(ORIGINAL_CMAKE_SIZEOF_VOIDP ${CMAKE_SIZEOF_VOIDP})
+set(CMAKE_SIZEOF_VOIDP "")
+write_basic_package_version_file(
+    "${PROJECT_BINARY_DIR}/${PORT_NAME}-version.cmake"
+    VERSION 0.0.1
```

Comment:
I haven't made any numbered releases of Flux, so I'm not sure where this version number comes from -- is it actually needed?

Reply:
You know, I don't think VERSION used to be required but I got a failure without it. I didn't try hard to not have one though. I'll look into it.

Reply:
The docs say

> [If no VERSION is given, the PROJECT_VERSION variable is used. If this hasn’t been set, it errors out.](https://cmake.org/cmake/help/v3.3/module/CMakePackageConfigHelpers.html#command:write_basic_package_version_file)

So, unfortunately, `VERSION` is actually needed.

I will set it to "-1" so even version 0 will be greater than it. Poking in the CMake source makes me believe that "-1" is valid where something like "(unset)" (with the "(" lexicographically less than "0") will fail do to CMake `math()` calls on it.